### PR TITLE
Refactor: Inflation logic + Settings back stack fix

### PR DESCRIPTION
**Pull request by BradleyWire**
## What's new in this PR?

Fix some issues with the back stack when you press back on some of the newer screens and refactored some of the inflation logic to use fragment constructors rather than having a single `onCreateView` method. 
#### APK
[Download build #1245](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1245/artifact/build/artifact/wire-dev-PR2608-1245.apk)
[Download build #1246](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1246/artifact/build/artifact/wire-dev-PR2608-1246.apk)

### CODE REVIEW

File: app/src/main/kotlin/com/waz/zclient/settings/account/editphonenumber/CountryCodePickerFragment.kt [hunk 1]
```diff
@@ -32,6 +32,11 @@ class CountryCodePickerFragment : DialogFragment() {
 
     private var countryCodeListener: CountryCodePickerListener? = null
 
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setStyle(STYLE_NO_FRAME, R.style.Theme_Dark_Preferences)
```
Comment (gizemb) [inline app/src/main/kotlin/com/waz/zclient/settings/account/editphonenumber/CountryCodePickerFragment.kt]: why do we force dark theme?🤔

Comment (mejdoo) [reply app/src/main/kotlin/com/waz/zclient/settings/account/editphonenumber/CountryCodePickerFragment.kt]: I guess this is the actual implementation as well in the scala code

Comment (BradleyWire) [reply app/src/main/kotlin/com/waz/zclient/settings/account/editphonenumber/CountryCodePickerFragment.kt]: This is how it's implemented in the current screen for the edit phone journey in Scala for other dialogs/screens. I'm sure we can look at changing it to be dynamic though.

Comment (gizemb) [reply app/src/main/kotlin/com/waz/zclient/settings/account/editphonenumber/CountryCodePickerFragment.kt]: ok then

**Pull request merged**